File: app/src/main/java/com/ashlikun/media/simple/data/HuoShanData.java
package com.ashlikun.media.simple.data;

import com.ashlikun.okhttputils.http.response.HttpResponse;
import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * 作者　　: 李坤
 * 创建时间: 2018/8/20　15:48
 * 邮箱　　：496546144@qq.com
 * <p>
 * 功能介绍：
 */
public class HuoShanData extends HttpResponse {
    public int result;
    public String pcursor;
    public int new_notify;
    public String llsid;
    @SerializedName("host-name")
    public String hostname;
    public List<FeedsData> feeds;

    public static class FeedsData {

        public ExtParamsData ext_params;

        public List<HeadurlsData> headurls;
        public String caption;
        @SerializedName(value = "cover_urls", alternate = {"main_mv_urls"})
        public List<AudioUrlsData> main_mv_urls;
        public List<AudioUrlsData> cover_thumbnail_urls;
        public List<AudioUrlsData> main_mv_urls_h265;

        public String getText() {
            return caption;
        }

        public static class ExtParamsData {
            public int mtype;
            public String color;
            public int w;
            public int sound;
            public int h;
            public int interval;
            public int video;
        }


        public int getWidth() {
            return ext_params != null ? ext_params.w : -2;
        }

        public String getImageUrl() {
            return (cover_thumbnail_urls != null && !cover_thumbnail_urls.isEmpty()) ? cover_thumbnail_urls.get(0).url : "";
        }

        public String getUrl() {
            return main_mv_urls != null ? main_mv_urls.get(0).url : "";
        }

        public float getHeight() {
            return ext_params != null ? ext_params.h : -2;
        }
    }

    public static class HeadurlsData {
        public String cdn;
        public String url;
        public String urlPattern;
    }

    public static class AudioUrlsData {
        public String cdn;
        public String url;
    }
}

File: app/src/main/java/com/ashlikun/media/simple/data/HuoShanData.kt
package com.ashlikun.media.simple.data

import com.ashlikun.okhttputils.http.response.HttpResponse
import com.google.gson.annotations.SerializedName

/**
 * 作者　　: 李坤
 * 创建时间: 2018/8/20　15:48
 * 邮箱　　：496546144@qq.com
 *
 *
 * 功能介绍：
 */
class HuoShanData : HttpResponse() {
    var result = 0
    var pcursor: String? = null
    var new_notify = 0
    var llsid: String? = null

    @SerializedName("host-name")
    var hostname: String? = null
    var feeds: List<FeedsData>? = null

    class FeedsData {
        var ext_params: ExtParamsData? = null
        var headurls: List<HeadurlsData>? = null
        var text: String? = null

        @SerializedName(value = "cover_urls", alternate = ["main_mv_urls"])
        var main_mv_urls: List<AudioUrlsData>? = null
        var cover_thumbnail_urls: List<AudioUrlsData>? = null
        var main_mv_urls_h265: List<AudioUrlsData>? = null

        class ExtParamsData {
            var mtype = 0
            var color: String? = null
            var w = 0
            var sound = 0
            var h = 0
            var interval = 0
            var video = 0
        }

        val width: Int
            get() = if (ext_params != null) ext_params!!.w else -2
        val imageUrl: String?
            get() = if (cover_thumbnail_urls != null && !cover_thumbnail_urls!!.isEmpty()) cover_thumbnail_urls!![0].url else ""
        val url: String?
            get() = if (main_mv_urls != null) main_mv_urls!![0].url else ""
        val height: Float
            get() = if (ext_params != null) ext_params!!.h.toFloat() else -2f
    }

    class HeadurlsData {
        var cdn: String? = null
        var url: String? = null
        var urlPattern: String? = null
    }

    class AudioUrlsData {
        var cdn: String? = null
        var url: String? = null
    }
}